src/components/popover-container/popover-container.component.tsx

```diff
@@ -168,6 +169,33 @@ export const PopoverContainer = ({
       setTimeout(() => closeButtonRef.current?.focus(), 0);
   }, [isOpen]);
 
+  const handleEscKey = useCallback(
```

Comment:
**suggestion**: this is mostly/entirely doing the same thing as the `handleCloseButtonClick` function that we already have. Unless you can think of a good reason not to I'd want to extract it to a common function (called something simple like `closePopover` and just call that from both places).

The one difference that I can find in current behaviour is that in your version the Escape press doesn't refocus the close button, whereas clicking the close icon (including via the return key) does. I really think this refocus behaviour should happen with escape too, so that's another good reason for reusing the code imo.

Reply:
That seems reasonable. I can certainly do that 👍🏻